prime-angular-frontend/src/app/modules/health-auth-site-reg/pages/site-management-page/site-management-page.component.ts

```diff
@@ -82,11 +105,62 @@ export class SiteManagementPageComponent implements OnInit {
   }
 
   public ngOnInit(): void {
+    this.createFormInstance();
+    this.initForm();
     const authorizedUser = this.authorizedUserService.authorizedUser;
     this.healthAuthorityId = authorizedUser.healthAuthorityCode;
-    this.healthAuthoritySites$ = this.authorizedUserResource.getAuthorizedUserSites(authorizedUser.id);
+    this.authorizedUserResource.getAuthorizedUserSites(authorizedUser.id)
+      .subscribe((sites: HealthAuthoritySiteList[]) => {
+        this.healthAuthoritySites = sites.sort((a, b) => a.siteName.toLocaleLowerCase().localeCompare(b.siteName.toLocaleLowerCase()))
```

Comment:
A site in progress may not have a Site Name yet

Reply:
Good point. Fixed.